chronos/asyncsync.nim

```diff
@@ -571,8 +571,9 @@ proc emit[T](bus: AsyncEventBus, event: string, data: T, loc: ptr SrcLoc) =
 
     # Schedule subscriber's callbacks, which are subscribed to the event.
     for subscriber in item.subscribers:
+      let subscriberCopy = subscriber
```

Comment:
What the purpose of this changes? `items()` iterator already makes copy of every returned item...

Reply:
`items` does not copy anymore in 1.4 and later.

Reply:
@Araq what is the intended semantic then, for lent for loop instances in local callbacks / closures? depending on the lifetime of the closure, a copy should or should not be made, for optimal performance

Reply:
Currently you cannot access a borrowed location via a closure. The compiler enforces the explicit copy here.